### PR TITLE
Add documentation on parameter comparators

### LINKED ISSUE #1588: Add documentation on parameter comparators
This issue follows up on the discussion between @hakonsbm and @stinebuu in #1567. 

The parameter comparators should be documented somewhere, possibly in the "NEST 3.0: Detailed transition guide".

### PR DESCRIPTION
**Pull request by hakonsbm**
Fixes #1588.

### CODE REVIEW

**Review by stinebuu — COMMENTED** [review]
Thanks, I have some questions, they might be really obvious, but I was confused :sun_with_face:

**Comment by hakonsbm** [comment]
@stinebuu I've reworded the sentence. Does it make more sense now?

**Review by stinebuu — APPROVED** [review]

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst [hunk 1]
```diff
@@ -406,6 +406,28 @@ statement. Three arguments are required:
...
+with another, and all comparison operators are supported. The result of such comparisons
+is a new ``nest.Parameter``, which evaluates to either 1 or 0 for true and false,
+respectively. The resulting comparison ``nest.Parameter`` can be used in a ``nest.logic.conditional()``,
+but also without it if the desired result is zero where the comparison is false.
+
+::
+
+    # As an example, take a step function where the resulting value is
+    # 0.5 for positive values, and zero for negative.
+    nest.logic.conditional(nest.random.uniform(min=-1., max=1.) > 0., 0.5, 0.)
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:425]:
```suggestion
    nest.logic.conditional(nest.random.uniform(min=-1., max=1.) > 0.0, 0.5, 0.0)
```

I know you don't have to add a zero after the decimal, but I think it looks strange especially being followed by a comma

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst [hunk 2]
```diff
@@ -406,6 +406,28 @@ statement. Three arguments are required:
...
+
+Compare parameters
+^^^^^^^^^^^^^^^^^^
+
+In the ``nest.logic.conditional()`` function above we used a comparison between a
+``nest.Parameter`` and a value. It is also possible to compare one ``nest.Parameter``
+with another, and all comparison operators are supported. The result of such comparisons
+is a new ``nest.Parameter``, which evaluates to either 1 or 0 for true and false,
+respectively. The resulting comparison ``nest.Parameter`` can be used in a ``nest.logic.conditional()``,
+but also without it if the desired result is zero where the comparison is false.
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:419]:
```suggestion
You can omit the ``nest.logic.conditional()`` if the desired result is zero when the comparison is false.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst [hunk 3]
```diff
@@ -406,6 +406,28 @@ statement. Three arguments are required:
...
+.. _compare_ex:
+
+Compare parameters
+^^^^^^^^^^^^^^^^^^
+
+In the ``nest.logic.conditional()`` function above we used a comparison between a
+``nest.Parameter`` and a value. It is also possible to compare one ``nest.Parameter``
+with another, and all comparison operators are supported. The result of such comparisons
+is a new ``nest.Parameter``, which evaluates to either 1 or 0 for true and false,
+respectively. The resulting comparison ``nest.Parameter`` can be used in a ``nest.logic.conditional()``,
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:418]:
```suggestion
respectively. The resulting comparison ``nest.Parameter`` can be used in a ``nest.logic.conditional()``.
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst [hunk 4]
```diff
@@ -406,6 +406,28 @@ statement. Three arguments are required:
...
+but also without it if the desired result is zero where the comparison is false.
+
+::
+
+    # As an example, take a step function where the resulting value is
+    # 0.5 for positive values, and zero for negative.
+    nest.logic.conditional(nest.random.uniform(min=-1., max=1.) > 0., 0.5, 0.)
+
+    # This comparison can be used directly without the nest.logic.conditional() function.
+    0.5*(nest.random.uniform(min=-1., max=1.) > 0.)
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:428]:
```suggestion
    0.5*(nest.random.uniform(min=-1., max=1.) > 0.0)
```

File: doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst [hunk 5]
```diff
@@ -406,6 +406,28 @@ statement. Three arguments are required:
...
+respectively. The resulting comparison ``nest.Parameter`` can be used in a ``nest.logic.conditional()``,
+but also without it if the desired result is zero where the comparison is false.
+
+::
+
+    # As an example, take a step function where the resulting value is
+    # 0.5 for positive values, and zero for negative.
+    nest.logic.conditional(nest.random.uniform(min=-1., max=1.) > 0., 0.5, 0.)
+
+    # This comparison can be used directly without the nest.logic.conditional() function.
```
Comment (jessica-mitchell) [inline doc/userdoc/guides/nest2_to_nest3/nest3_features/parametrization.rst:427]:
```suggestion
    # This comparison can be used without the nest.logic.conditional() function.
```

**Review by jessica-mitchell — CHANGES_REQUESTED** [review]
Looks good, just a few minor suggestions

**Review by jessica-mitchell — APPROVED** [review]
Thanks @hakonsbm!

**Pull request merged**